internal/database/own_signal_configurations.go

```diff
@@ -42,8 +44,22 @@ func (s *signalConfigurationStore) With(other basestore.ShareableStore) *signalC
 	return &signalConfigurationStore{s.Store.With(other)}
 }
 
-func (s *signalConfigurationStore) LoadConfigurations(ctx context.Context) ([]SignalConfiguration, error) {
-	q := "SELECT id, name, description, excluded_repo_patterns, enabled FROM own_signal_configurations ORDER BY id;"
+type LoadSignalConfigurationArgs struct {
+	Name string
+}
+
+func (s *signalConfigurationStore) LoadConfigurations(ctx context.Context, args LoadSignalConfigurationArgs) ([]SignalConfiguration, error) {
+	q := "SELECT id, name, description, excluded_repo_patterns, enabled FROM own_signal_configurations %s ORDER BY id;"
+
+	var conds []*sqlf.Query
+	if len(args.Name) > 0 {
+		conds = append(conds, sqlf.Sprintf("name = %s", args.Name))
+	}
+	where := sqlf.Sprintf("")
+	if len(conds) > 0 {
+		where = sqlf.Sprintf("WHERE %s", sqlf.Join(conds, "AND"))
+	}
```

Comment:
There seems to be at most one cond. Or otherwise to make the above generate correct SQL I think `"AND"` needs to become `" AND "` with whitespace around. I don't believe that sqlf is smart enough to do that.

```suggestion
	where := sqlf.Sprintf("")
	if len(args.Name) > 0 {
		where = sqlf.Sprintf("WHERE name = %s", args.Name)
	}
```